### ISSUE TITLE
Deprecate manylinux 2010 

### ISSUE THREAD

**Issue by freddyaboulton** [alan-turing-institute/sktime#1377]
**Is your feature request related to a problem? Please describe.**

We currently support manylinux 2010 wheels. However, manylinux2010 reached its end-of-life last year and our dependencies are no longer supporting it going forward. 

Two examples so far:
1. llvmlite/numba stopped supporting manylinux 2010: https://github.com/numba/llvmlite/issues/768
2. Pillow released version 8.3.2 today that no longer supports manylinux 2010: https://pypi.org/project/Pillow/#files . This is breaking our CI

**Describe the solution you'd like**
Drop support for manylinux 2010 so we can use the latest versions of our dependencies 

**Describe alternatives you've considered**
Pin our dependencies to the last version that supports manylinux 2010. Or use separate dependencies only for manylinux2010.

**Additional context**

**Comment by mloning** [comment]
2. seems to be resolved, CI is working again on `main`

**Comment by fkiraly** [comment]
well - should we wait with deprecation then?

**Comment by freddyaboulton** [comment]
I'm ok with waiting to do this until the 0.8.0 release (btw when will that be?) as @mloning seems to have indicated by switching the milestone on https://github.com/alan-turing-institute/sktime/pull/1379

**Comment by mloning** [comment]
Yes, we haven't established a clear release schedule but I would like to aim for a release every month, we're currently a bit behind schedule, also due to the summer break, but hopefully can push out more regular releases soon. Any preference from your side @freddyaboulton?

**Comment by mloning** [comment]
I've now tentatively scheduled v0.8.0 for the beginning of November (see https://github.com/alan-turing-institute/sktime/milestone/8). After I'd like to try to make a new release at the beginning of every month.

**Comment by freddyaboulton** [comment]
Sounds good to me @mloning !
